src/widget/weffectselector.cpp

```diff
@@ -44,14 +44,14 @@ void WEffectSelector::populate() {
     blockSignals(true);
     clear();
 
-    // TODO(xxx): filter out blacklisted effects
-    // https://bugs.launchpad.net/mixxx/+bug/1653140
     const QList<EffectManifestPointer> availableEffectManifests =
             m_pEffectsManager->getAvailableEffectManifests();
     QFontMetrics metrics(font());
 
     for (int i = 0; i < availableEffectManifests.size(); ++i) {
         const EffectManifestPointer pManifest = availableEffectManifests.at(i);
+        if (!pManifest->isVisible())
+            continue;
```

Comment:
I don't think WEffectSelector should be responsible for determining which effects to show. Calling `m_pEffectsManager->getAvailableEffectManifests()` above should take care of that for WEffectSelector. I think the only change needed to WEffectSelector is to call `WEffectSelector::populate` again when a signal is received from changing the blacklisted effects in the preferences.